GetFileSystemEntries returns an array of full paths of the directory contents. But GetCurrentSegmentGeneration expects an array of only the names of the directory contents. This has been reported upstream.

From: Debajyoti Bera

Index: Index/IndexReader.cs
===================================================================
--- Index/IndexReader.cs	(revision 4506)
+++ Index/IndexReader.cs	(working copy)
@@ -405,7 +405,12 @@
 		
 		public static bool IndexExists(System.IO.FileInfo directory)
 		{
-			return SegmentInfos.GetCurrentSegmentGeneration(System.IO.Directory.GetFileSystemEntries(directory.FullName)) != - 1;
+			string[] files = System.IO.Directory.GetFileSystemEntries(directory.FullName);
+                        for (int i = 0; i < files.Length; i++)
+                        {
+			    files[i] = System.IO.Path.GetFileName (files[i]);
+                        }
+			return SegmentInfos.GetCurrentSegmentGeneration(files) != - 1;
 		}
 		
 		/// <summary> Returns <code>true</code> if an index exists at the specified directory.
@@ -1017,4 +1022,4 @@
 			}
 		}
 	}
-}
\ No newline at end of file
+}
